opm/parser/eclipse/EclipseState/Summary/Summary.cpp

```diff
@@ -79,30 +82,114 @@ namespace Opm {
         return res;
     }
 
+    static inline std::array< int, 3 > dimensions( const EclipseGrid& grid ) {
+        return {
+            int( grid.getNX() ),
+            int( grid.getNY() ),
+            int( grid.getNZ() )
+        };
+    }
+
+    static inline std::array< int, 3 > getijk( const DeckRecord& record,
+                                               int init = 0 )
+    {
+        return {
+            record.getItem( init + 0 ).get< int >( 0 ) - 1,
+            record.getItem( init + 1 ).get< int >( 0 ) - 1,
+            record.getItem( init + 2 ).get< int >( 0 ) - 1
+        };
+    }
+
+    static inline std::array< int, 3 > getijk( const Completion& completion ) {
+        return { completion.getI(), completion.getJ(), completion.getK() };
+    }
+
     static inline std::vector< ERT::smspec_node > keywordB(
             const DeckKeyword& keyword,
             const EclipseState& es ) {
 
-        std::array< int, 3 > dims = {{
-            int( es.getEclipseGrid()->getNX() ),
-            int( es.getEclipseGrid()->getNY() ),
-            int( es.getEclipseGrid()->getNZ() )
-        }};
+        auto dims = dimensions( *es.getEclipseGrid() );
 
         const auto mkrecord = [&dims,&keyword]( const DeckRecord& record ) {
-
-            std::array< int , 3 > ijk = {{
-                record.getItem( 0 ).get< int >( 0 ) - 1,
-                record.getItem( 1 ).get< int >( 0 ) - 1,
-                record.getItem( 2 ).get< int >( 0 ) - 1
-            }};
-
+            auto ijk = getijk( record );
             return ERT::smspec_node( keyword.name(), dims.data(), ijk.data() );
         };
 
         return fun::map( mkrecord, keyword );
     }
 
+    static inline std::vector< ERT::smspec_node > keywordR(
+            const DeckKeyword& keyword,
+            const EclipseState& es ) {
+
+        auto dims = dimensions( *es.getEclipseGrid() );
+
+        const auto mknode = [&dims,&keyword]( int region ) {
+            return ERT::smspec_node( keyword.name(), dims.data(), region );
+        };
+
+        const auto& item = keyword.getDataRecord().getDataItem();
+        const auto regions = item.size() > 0 && item.hasValue( 0 )
+            ? item.getData< int >()
+            : es.getRegions();
+
+        return fun::map( mknode, regions );
+    }
+
+   static inline std::vector< ERT::smspec_node > keywordC(
+           const DeckKeyword& keyword,
+           const EclipseState& es ) {
+
+       std::vector< ERT::smspec_node > nodes;
+       const auto& keywordstring = keyword.name();
+       const auto& schedule = es.getSchedule();
+       const auto last_timestep = schedule->getTimeMap()->last();
+       auto dims = dimensions( *es.getEclipseGrid() );
+
+       for( const auto& record : keyword ) {
+
+           const auto& wellname = record.getItem( 0 ).get< std::string >( 0 );
+
+           if( wellname == "*" ) {
```

Comment:
We should have `well_item.defaultApplied( 0 ) == true` *both* for empty input and for a literal '*'; see e.g.: https://github.com/OPM/opm-parser/blob/e9e552141f102c7689e8c02e65a2ad86c86db7b6/opm/parser/eclipse/IntegrationTests/ParseDATAWithDefault.cpp#L88

> It wasn't empty in the input.


Empty input is actually not an important concept in this context - the question is really:
```C++
if item.defaultApplied( n ) ...
```
and that will not differentiate between empty input and literal `*` - on the other hand the `getValue( )` methods will not differentiate between values injected in as defaults and values entered explicitly in the deck.